extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryManager.java

```diff
@@ -0,0 +1,262 @@
+        lock.lock();
+        try {
+            LOG.info("Starting Query Manager.");
+            queryExecutor.startAndWait();
+            source.startAndWait();
+
+            // subscribe to the sources to be notified of changes.
+            source.subscribe(new QueryManagerSourceListener());
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    @Override
+    protected void shutDown() throws Exception {
+        lock.lock();
+        try {
+            LOG.info("Stopping Query Manager.");
+            source.stopAndWait();
+            queryExecutor.stopAndWait();
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    /**
+     * An implementation of {@link QueryChangeLogListener} for the
+     * {@link QueryManager}.
+     * <p>
+     * When notified of a {@link ChangeType} performs one of the following:
+     * <li>{@link ChangeType#CREATE}: Creates a new query using the
+     * {@link QueryExecutor} provided to the {@link QueryManager}</li>
+     * <li>{@link ChangeType#DELETE}: Deletes a running query by stopping the
+     * {@link QueryExecutor} service of the queryID in the event</li>
+     * <li>{@link ChangeType#UPDATE}: If the query is running and the update is
+     * to stop the query, stops the query. Otherwise, if the query is not
+     * running, it is removed.</li>
+     */
+    private class QueryManagerQueryChange implements QueryChangeLogListener {
+        private final String ryaInstanceName;
+
+        /**
+         * Creates a new {@link QueryManagerQueryChange}.
+         *
+         * @param ryaInstanceName - The rya instance the query change is
+         *        performed on. (not null)
+         */
+        public QueryManagerQueryChange(final String ryaInstanceName) {
+            this.ryaInstanceName = requireNonNull(ryaInstanceName);
+        }
+
+        @Override
+        public void notify(final ChangeLogEntry<QueryChange> queryChangeEvent, final Optional<StreamsQuery> newQueryState) {
+            LOG.debug("New query change event.");
+            final QueryChange entry = queryChangeEvent.getEntry();
+
+            lock.lock();
+            try {
+
+                switch (entry.getChangeType()) {
+                    case CREATE:
+                        if(newQueryState.isPresent()) {
+                            runQuery(ryaInstanceName, newQueryState.get());
+                        } else {
+                            LOG.error("The query with ID: " + entry.getQueryId() + " must be present with the change to be created.");
+                        }
+                        break;
+                    case DELETE:
+                        stopQuery(entry.getQueryId());
+                        break;
+                    case UPDATE:
+                        if (!newQueryState.isPresent()) {
+                            LOG.debug("The query with ID: " + entry.getQueryId() + " must be provided with the update, cannot perform update.");
```

Comment:
This log doesn't reflect your is present check.

Reply:
done